test-suite/djinni/extended_record.djinni

```diff
@@ -0,0 +1,5 @@
+
+# Extended record
+extended_record = record +c {
+    foo: bool;
+}
```

Comment:
Funny story, but I just tried this and I can't.  Creating a constant then forces the code to generate a .cpp file inside the generated-src/cpp folder, and that file includes the subclass with an include of the form: #include "../extended_record.hpp" which can't be found if I put the handwritten source with the other handwritten source.  If we wait until my other PR is pulled I could probably get this additional test to work.